Fix and update solid-body cs32 experiment

jm-c opened the pull request:
## What changes does this PR introduce?
Update experiment 'solid-body.cs-32x32x1`:
1. fix initial condition for initial surf. pressure (see issue #793)
2. change flow speed, solid-sphere rotation and radius of sphere ;
3. switch to "compact format" (to get at least 1 experiment to use this).

## What is the current behaviour? 
see issue #793
also no alternative for providing initial conditions from files (+ gendata was not up to date)

## What is the new behaviour 
updated and fixed.
and `gendata.m` allows to write initial condition files as an alternative to customized src code.

## Does this PR introduce a breaking change? 
no 

## Other information:

## Suggested addition to `tag-index`
o verification/solid-body.cs-32x32x1:
- update set up (faster flow, smaller planet ..) and fix initial Ps conditions .

jm-c commented:
I think this PR is ready for review.
I've just updated the customized `ini_psurf.F` to remove the global-mean value from initial `Eta` (also added `rhoConst` to have correct units) so that now we get a good agreement when comparing with same test but using the 3 bin files that `gendata.m` can produce for initial conditions.
I could think of adding a short matlab script to plot the difference in Eta after 25.iter, but I would use `grph_CS.m` (from `utils/matlab/cs-grid/bk_line/`) for that and it does not work until PR #792 is meeged in, so probably better to leave this for a future PR.

jrscott commented:
@jm-c looked a bit at this


Suggest adding a README here: mention compact format used, basic explanation of the setup, see additional explanations in gendata.m etc. Is there an easy place to point where the analytic expression comes from?

minor stuff:
- add comment in gendata.m where it uses split_Z_cub note where this can be found
- suggest make kwr=2 in gendata so by default it dumps all the .bin files
- line 65 in gendata s/b sin^2?
- what's rdwr_grid [and what's rdda())? What's TUV.mat?
- in gendata the (compact) format manipulations are obscure, but I think that's a different issue to be solved elsewhere
- put comment in data to uncomment the U_init etc lines once these files have been created

jm-c commented:
@jrscott Thanks for the review.

I added a `README.md` file with the analytical expression of initial wind and surface-pressure anomaly.

And regarding scpecific points:
>  add comment in gendata.m where it uses split_Z_cub note where this can be found

I added a comment at the end of README.    
> suggest make kwr=2 in gendata so by default it dumps all the .bin files
 
The purpose of `kwr`  = 2 vs =1 is explained in README
>  line 65 in gendata s/b sin^2?

I added a comment in README, section "Setup details"`
>   what's rdwr_grid [and what's rdda())? What's TUV.mat?

`rdwr_grid.m` is unaffected by this PR. It was the matlab script that was used to make these `tile00?.mitgrid` from a set of old grid-files (with 1 file per field). I prefer to keep it as the same 6 grid files are used here. And regarding `TUV.mat`, see issue #791 and #761.
>  in gendata the (compact) format manipulations are obscure, but I think that's a different issue to be solved elsewhere

I am going to add 1 line of comment in `gendata.m`; but I think that the compact-format does not require extra manipulation and instead it is the old-format that requires to apply some "permute" steps.

>  put comment in data to uncomment the U_init etc lines once these files have been created

This is explained in the README.

jrscott commented:
LGTM
